File: app/src/main/java/dev/alimansour/iweather/data/local/converters/Converters.java
package dev.alimansour.iweather.data.local.converters;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.util.List;

/**
 * WeatherApp Android Application developed by: Ali Mansour
 * ----------------- WeatherApp IS FREE SOFTWARE -------------------
 * https://www.alimansour.dev   |   mailto:dev.ali.mansour@gmail.com
 */
public class Converters {
    @TypeConverter
    public static List<String> restoreList(String listOfString) {
        return new Gson().fromJson(listOfString, new TypeToken<List<String>>() {
        }.getType());
    }

    @TypeConverter
    public static String saveList(List<String> listOfString) {
        return new Gson().toJson(listOfString);
    }
}

File: app/src/main/java/dev/alimansour/iweather/data/local/converters/Converters.kt
package dev.alimansour.iweather.data.local.converters

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

/**
 * WeatherApp Android Application developed by: Ali Mansour
 * ----------------- WeatherApp IS FREE SOFTWARE -------------------
 * https://www.alimansour.dev   |   mailto:dev.ali.mansour@gmail.com
 */
class Converters {
    @TypeConverter
    fun restoreList(listOfString: String?): List<String?>? =
        Gson().fromJson(listOfString, object : TypeToken<List<String?>?>() {}.type)

    @TypeConverter
    fun saveList(listOfString: List<String?>?): String? = Gson().toJson(listOfString)
}